File: b_rank_new_level_up_problems/b_rank_new_level_up_problems__get_richer_sushi.cs
// 実行時間 30ms
using static System.Console;
using static System.Linq.Enumerable;

internal class Program
{
    static void Main()
    {
        int o = 0;
        int[] n = ReadLine().Split().Select( int.Parse ).ToArray();
        int[] a = new int[n[0] * 2];

        foreach( int i in Range( 0, n[0] ) ) {
            int p = int.Parse( ReadLine() );

            a[i] = p;
            a[i + n[0]] = p;
        }
        foreach( int i in Range( 0, n[0] ) ) {
            int p = 0;

            foreach( int j in Range( i, n[1] ) ) {
                p += a[j];
            }
            o = System.Math.Max( o, p );
        }
        WriteLine( o );
        return;
    }
}

File: b_rank_new_level_up_problems/b_rank_new_level_up_problems__get_richer_sushi.py
# 実行時間 60ms
import sys

rl = sys.stdin.readline
m = 0
n, k = list(map(int, rl().split()))
a = [0 for _ in [0] * (n * 2)]

for i in range(n):
    p = int(rl())

    a[i] = p
    a[i + n] = p
for i in range(n):
    p = 0

    for j in range(i, i + k):
        p += a[j]
    m = max(m, p)
print(m)
